homeworks/03-data-structure/src/merge-sort/mergeSort.ts

```diff
@@ -0,0 +1,33 @@
+export type CompareFunction<T> = (a: T, b: T) => number;
+
+export type MergeSort = <T>(arr: T[], compare: CompareFunction<T>) => T[];
+
+export type MergeArrays = <T>(aArr: T[], bArr: T[], compare: CompareFunction<T>) => T[];
+
+export const mergeArrays: MergeArrays = (leftArr, rightArr, compare) => {
+  const arr = [];
```

Comment:
Always define type when declaring an empty array.

Reply:
I have done such way, since not sure how i can get correct return type from generic. 

```
export const mergeArrays: MergeArrays = (leftArr, rightArr, compare) => {
  const arr: typeof leftArr = [];
...
```

Reply:
Seems, this is only one option here - like you did.

But you don't need separate types for `MergeArrays` and `MergeSort`, moreover, you don't use it anywhere at all.
Just enough to write a function correctly typed and you can use generic type inside:
```ts
export function mergeArrays<T>(leftArray: T, rightArray: T, compare: CompareFunction<T>): T[] {
  const array: T[] = [];
  // ...
}
```

Reply:
Understand, rewritten - https://github.com/vbarbasins/evo-ts-bootcamp/pull/3/commits/54903ad8ae790219026d2721bc85c938300a38bf